string object_type
string object_id
 ---